docs/lifecycle-management.md

```diff
@@ -0,0 +1,185 @@
+# Higher level VM Management
+
+## Motivation
+
+So far KubeVirt provides a pod-like interface when managing VMs. This seems to
+be a good building block.
+On the other hand this might not be sufficient for usual virtualization
+management UIs, which do expect to manage the complete life-cycle of a VM,
+including shut down VMs 8which is currently not intended by the core runtime).
+
+
+This document outlines how minimal assumptions can be met and also visits how
+other requirements impact the design.
+
+
+## Requirements
+* Running and non-running VMs are handled through the same API
+* Creating a VM must be separate from launching a VM
+* Support for “flavor” based VMs (templating)
+* Support for expanding sparse VM definitions
+* Support for detailed VM definitions
+* Support hot-plugging
+* Representation of dynamic instance specific data
+* Support for defaulting (to support OS compatibility)
+Considerations
+* Sub-resources for actions
+* Presets to add certain configurations
+
+
+## Definitions
+* **Defaulting** - Setting certain defaults during expansion (i.e. virtio for a
+  bus type, or e1000 for as a nic model)
+* **Expansion** - VM definitions can be provided in a sparse way i.e. just a
+  disk. Once the VM is getting started, the runtime will infer additional
+  requirements for this device and add them to the runtime VM definition.
+  This process of inferring and adding additional devices is called expansion.
+* **Flavor** - Comparable to a template or a specific instance type
+* **Preset** - Presetting certain values in a specified path of the document
+  (i.e. setting .spec.domain.devices.nvram). It is not about setting certain
+  default values for undefined fields (i.e. set virtio for all busses)
```

Comment:
If we are talking about libosinfo like defaulting, I would see it as an extra step, which can be used by a management plane. However, there will always be the other defaulting, presumably after presets were applied, which just fills out everything left, to have a fully filled VM Spec. 

As I tend to repeat (;) ), I see there a possibility to let that last defaulting happen in a synchronous way, by using a libvirt instance combined with a guest machine type. While back-propagating status information, or even spec about current applied changes,  would still be possible, it would reduce the need to do so. Even better would be, if we could somehow document the defaults in the API. That is something I would normally expect from an application. By having a single point, which adds that to the spec (with different libvirts and qemus across the cluster, which might set  different defaults for the same fields), instead of letting nodes do that, that becomes possible.